### PR TITLE
Add a lockfile to avoid concurrent access

### PR DESCRIPTION
**Pull request by clementperon**
Doing concurrent acces on mender grub environement utility will trig the following error:
/usr/bin/grub-editenv: error: invalid environment block.

This append when a process is editing an env so the other is trying to restore it.

Fix this by running critical section under a flock

Signed-off-by: Clément Péron <peron.clem@gmail.com>

### CODE REVIEW

**Review by kacf — CHANGES_REQUESTED** [review]
Thanks, looks like a good fix. A few small fixups:

* Make sure the author address matches the signoff address (this is why the CI is failing).

File: grub-mender-grubenv-print [hunk 1]
```diff
@@ -57,6 +57,7 @@ LOCKSUM1="$ENV_DIR/mender_grubenv1/lock.sha256sum"
 ENV2="$ENV_DIR/mender_grubenv2/env"
 LOCK2="$ENV_DIR/mender_grubenv2/lock"
 LOCKSUM2="$ENV_DIR/mender_grubenv2/lock.sha256sum"
+LOCKFILE="/var/lock/fw_printenv.lock"
```
[the remark this answers is not in the document]
Comment (clementperon) [reply grub-mender-grubenv-print]: done

**Comment by clementperon** [comment]
@kacf thanks i have fixed the commit author email.

Do you want me to add an entry in the Changelog ?

**Comment by kacf** [comment]
No, you don't need to for this repository. I will add one once I update the SHA in meta-mender and mender-convert. The bot is just being a bit overeager (we will probably disable it for this repository).

**Review by kacf — APPROVED** [review]
Thanks!

**Comment by kacf** [comment]
Changes applied in https://github.com/mendersoftware/meta-mender/pull/1879 and https://github.com/mendersoftware/mender-convert/pull/526.

**Comment by kacf** [comment]
@clementperon: Looks like `sh` does not support the redirection syntax. Check out https://github.com/mendersoftware/grub-mender-grubenv/pull/29.

**Pull request merged**